traffic_router/core/src/test/java/org/apache/traffic_control/traffic_router/core/external/LocationsTest.java

```diff
@@ -139,4 +144,38 @@ public void itGetsCachesForALocation() throws Exception {
 			if (response != null) response.close();
 		}
 	}
+
+	@Test
+	public void itHandlesHeadRequests() throws Exception {
+		final List<String> paths = new ArrayList<String>();
+		paths.add("http://localhost:3333/crs/locations");
+		paths.add("http://localhost:3333/crs/locations/caches");
+
+		CloseableHttpResponse response = null;
+
+		try {
+			final HttpGet httpGet = new HttpGet("http://localhost:3333/crs/locations");
+			response = closeableHttpClient.execute(httpGet);
+
+			ObjectMapper objectMapper = new ObjectMapper(new JsonFactory());
+			JsonNode jsonNode = objectMapper.readTree(EntityUtils.toString(response.getEntity()));
+
+			String location = jsonNode.get("locations").get(0).asText();
+			paths.add("http://localhost:3333/crs/locations/" + location + "/caches");
+		} catch (Exception e) {
+			fail(e.getMessage());
+		} finally {
+			if (response != null) response.close();
+		}
+
+		for (final String path : paths) {
+			final HttpHead httpHead = new HttpHead(path);
+			try {
+				response = closeableHttpClient.execute(httpHead);
+				assertThat(response.getStatusLine().getStatusCode(), equalTo(200));
```

Comment:
Shouldn't you also check that it sets the Content-Length header?

Reply:
Checking that it sets the Content-Length header in 81bc50d2d0